### ISSUE TITLE
Please add compatibility for NC28

### ISSUE THREAD

**Issue by snagglewest** [nextcloud/files_texteditor#816]
Please add compatibility for NC28

**Comment by SteveDinn** [comment]
I have just come to NextCloud in v28. I cannot understand why they insist on treating all text as Markdown and force a WYSIWYG editor. I want to edit my Markdown as plain text, please!

**Comment by red3333** [comment]
Nothing but bot stuff for the last 7 months...
Is this project dead ?

**Comment by Polve** [comment]
I miss this app to update to v28 :-(

**Comment by inguy24** [comment]
I re-enabled the app on NC28 and it seems to be working fine for me. No log errors and it edits the files as before. Will monitor it for problems but seems like there may not be breaking changes in NC28 and the app just needs updated to flag it as compatible.

**Comment by elpraga** [comment]
Thank you for the update @inguy24 !

So are you saying tha it still appears in the menu and can it coexist with the official Text app?

**Comment by inguy24** [comment]
I am not sure what the official text app is, but yes, the app seems to function normally and is in the menu. Again, it is functioning for me with my configuration and will continue to monitor for issues. I can not guarantee it will work for everyone. 

I would experiment with enabling it in a test environment or, if no test environment, on a production server during a time that you can afford to have your system offline if things don't go well. Be familiar with the occ commands to disable the app if need be. 



⁣Get BlueMail for Android ​

**Comment by red3333** [comment]
So I re-tested with the latest Nextcloud 28.0.4:
1. I disabled the (official) "Text" app and "Plain text editor" app
2. I tried to open a .txt file (click on it), logically, as no text app is enabled on Nextcloud, the file is downloaded instead of opened.
3. I enabled "Plain text editor", then opened a .txt file, still downloaded
4. I enabled (official) "Text" app, then opened a .txt file, it gets opened for editing (in "Text" app, which I like less...)

I have the feeling that the "Plain text editor" app does no get registered as a text editor app in Nextcloud 28.

**Comment by elpraga** [comment]
That behavior would be quite desirable as long as the option to open with
plain text editor remains accessible through the menu (hence the question
on my previous post)

Those who only want to use the Plain Text editor app can uninstall the Text
app.

Those who do not want to use the Plan Text app do not install it at all

And those (like me) who would use the Text app for most files, but do need
the ability to edit some text files directly can do so using the menu and
choosing the Plain Text app.

**Comment by elpraga** [comment]
Ahh, sorry, it's morning here, and my eyes haven't opened properly yet...

I missed the **"still downloaded"** part.

Could you please check if files can be edited using the Plain Text editor thorough the menu?

**Comment by red3333** [comment]
> So I re-tested with the latest Nextcloud 28.0.4:
> 
>     1. I disabled the (official) "Text" app and "Plain text editor" app
> 
>     2. I tried to open a .txt file (click on it), logically, as no text app is enabled on Nextcloud, the file is downloaded instead of opened.
> 
>     3. I enabled "Plain text editor", then opened a .txt file, still downloaded
> 
>     4. I enabled (official) "Text" app, then opened a .txt file, it gets opened for editing (in "Text" app, which I like less...)
> 
> 
> I have the feeling that the "Plain text editor" app does no get registered as a text editor app in Nextcloud 28.

Going back to step 3., there is no option in the menu for editing the file (while in step 4. the option is present and opens with (official) "Text" app)

**Comment by gohrner** [comment]
I also tried to just "force enable" it, and there's no text editing menu entry for me in NC 28 either, unfortunately. :-(

**Comment by haohetao** [comment]
Please add compatibility for NC28

**Comment by ernolf** [comment]
Hi @icewind1991 

> I've started a basic POC but probably don't have the time to finish it

How concrete is this splendid idea in the meantime? Can we hope for a replacement solution for the plain text editor or is that still up in the air?

ernolf

**Comment by workingmachine** [comment]
Please add compatibility for NC28.

**Comment by jahway603** [comment]
> By this point, it might be better to add support for NC29.

Would that also add support for NC28 at the same time?

**Comment by max-nextcloud** [comment]
> I have just come to NextCloud in v28. I cannot understand why they insist on treating all text as Markdown and force a WYSIWYG editor. I want to edit my Markdown as plain text, please!

Sorry if this is off topic here with respect to NC28 compatibility. But you can switch to plain text editing for markdown with a app config setting:
```
 occ config:app:set text rich_editing_enabled --value="0" 
```

**Comment by daffydock** [comment]
> > I have just come to NextCloud in v28. I cannot understand why they insist on treating all text as Markdown and force a WYSIWYG editor. I want to edit my Markdown as plain text, please!
> 
> Sorry if this is off topic here with respect to NC28 compatibility. But you can switch to plain text editing for markdown with a app config setting:
> 
> ```
>  occ config:app:set text rich_editing_enabled --value="0" 
> ```

Indeed, a bit off topic, true, but still very relevant and useful. Thank you @max-nextcloud for sharing. For the time being, applying this change brings me back to where I want to be until an update comes. Plus, it makes the Text app run a bit faster, so if it can work, even as a stopgap for others, then I recommend to give it a go.  
Cheers.

**Comment by SteveDinn** [comment]
IMHO, the best implementation of a markdown editor is in the Notes app. It allows you to switch back and forth between editing (markdown, but a kind of WYSIWIG markdown) and viewing, where it shows everything without the markdown symbols.

The editor in the default text editor is the worst out of all of them. I often find it very hard to get it to do what I want. Setting that `rich_editing_enabled` config option to 0 doesn't allow you to see the markdown in "viewing" mode at all anymore.

**Comment by Sariyn** [comment]
Not to be rude but decisions like how markdown was handled in v28 is what gives open source such a bad name sometimes. It simply defies commonsense. Please do keep in mind that people who use software like this may tend towards being power users, and would like to edit markdown manually. Please reinstate support for this obvious feature of using markdown.

**Comment by SteveDinn** [comment]
It would be great if we could have a default app for each file type and *also* an "Open with..." menu that would allow selection of any app compatible with the file type. This way,.for markdown files, we could have the current markdown editor for whomever likes it, also a plain text editor, and even (my personal favourite) the markdown editor from the Notes app
